tasks/post_to_queue_and_trigger_step_function/sqs_library.py

```diff
@@ -0,0 +1,140 @@
+"""
+Name: sqs_library.py
+Description: library for post_to_queue_and_trigger_step_function lambda function for posting to fifo SQS queue.
+Largely copied from copy_to_glacier
+"""
+# todo: Move to shared lib
+# Standard libraries
+import functools
+import hashlib
+import json
+import random
+import time
+from typing import Any, Callable, Dict, TypeVar
+
+# Third party libraries
+import boto3
+import fastjsonschema
+from cumulus_logger import CumulusLogger
+
+# Set Cumulus LOGGER
+LOGGER = CumulusLogger(name="ORCA")
+MAX_RETRIES = 3  # number of times to retry.
+BACKOFF_FACTOR = 2  # Value of the factor used to backoff
+INITIAL_BACKOFF_IN_SECONDS = 1  # Number of seconds to sleep the first time through.
+RT = TypeVar("RT")  # return type
+
+
+# Retry decorator for function
+# todo: Untested. Will likely eventually be part of shared lib.
```

Comment:
Linked task for shared lib.

I don't understand "not sure `untested` is fair here." It is untested in the code I pulled this from.
If you are suggesting that it counts as `tested` because the code is similar to other tested code, then I will push back against that.

Reply:
I guess I was referring to if you have unit tests, verifying the logic and you have run basic validation tests on the code in AWS to validate the code (which could be done by passing an invalid SQS URL), it is no longer `untested`.

Reply:
It is not covered by unit tests. Refined comment.